### ISSUE TITLE
MA0150 false positive when a sealed type inherits a ToString override from a base class

### ISSUE THREAD

**Issue by BenGearset** [meziantou/Meziantou.Analyzer#1270]
### Version of the Meziantou.Analyzer NuGet package

3.0.145

### Rule Identifier

MA0150

### Target Framework

net10.0

### C# Language version

C# 14

### .NET SDK version

10.0.100

### Description

Since 3.0.143, MA0150 reports sealed types that inherit `ToString` from a base class. The inherited override runs at runtime, not `object.ToString`. 3.0.142 does not report these cases.

The rule also fires on BCL types. `HttpResponseHeaders` is sealed and inherits its override from `HttpHeaders`

### Reproduction Steps

Minimal code:

```c#
class Base
{
    public override string ToString() => "value";
}

sealed class Derived : Base
{
}

var s = $"{new Derived()}"; // MA0150: ToString on 'Derived' will use the default object.ToString
```
